Data_Package_Documentation/IGSN_Formatting/recreate_SESAR_template.R

```diff
@@ -0,0 +1,393 @@
+  # fix columns.
+
+# Objective: Recreate a SESAR upload template from samples already registered in the SESAR database
+  # Because we can't provide the original SESAR upload template that we used to
+  # register samples, this script recreates the template and includes it in an 
+  # `ESS_DIVE_Infrastructure_ONLY` folder in the data package.
+
+# Assumptions: 
+  # Requires input IGSN file to have 2 columns: `IGSN` that has the sample IGSNs and `Parent_IGSN` that has the site IGSNs.
+  # The input file IGSNs should not have the full DOI URL. They can look like "10.58052/IEPRS007S" or "IEPRS007S".
+  # If there are more than 100 samples or sites, you will have to repeat the downloading process (but this script will walk you through that).
+
+# Directions: 
+  # Go through this script line by line for directions on how to navigate
+  # downloading info from the SESAR database and creating the psuedo template
+  # files
+
+# Status: 
+  # complete. 
+  # script has not yet been tested when there are more than 100 sites or samples
+
+
+### Prep Script ################################################################
+
+# load libraries
+library(tidyverse)
+library(devtools)
+library(rlog)
+library(clipr)
+library(readxl)
+library(testthat)
+
+# load functions
+source_url("https://raw.githubusercontent.com/river-corridors-sfa/rcsfa-data_processing_for_publication/refs/heads/main/Data_Transformation/functions/rename_column_headers.R")
+
+# user inputs - update these each time you run the script
+data_package_igsns <- read_csv("Z:/00_ESSDIVE/01_Study_DPs/SFA_SpatialStudy_2021_SampleData_v3/v3_SFA_SpatialStudy_2021_SampleData/SPS_Sample_IGSN-Mapping.csv", skip = 1) # read in the file from your data package that has IGSN and Parent_IGSN cols
+ess_dive_infrastructure_only_dir <- "Z:/00_ESSDIVE/01_Study_DPs/SFA_SpatialStudy_2021_SampleData_v3/ESS_DIVE_Infrastructure_ONLY"
+
+# create and open your temp directory - this folder and files in it will automatically delete when your R session ends
+temp_dir <- tempdir()
+shell.exec(temp_dir)
+
+
+### Download from SESAR ########################################################
+# first pull the IGSNs from the data package
+# manipulate them into a version that can be copied into the search query in SESAR
+# search for files
+# download them to a temporary directory
+
+# get site IGSN col
+site_igsns <- data_package_igsns %>% 
+  mutate(Parent_IGSN = str_replace_all(Parent_IGSN, "\\s+", "")) %>% 
+  select(Parent_IGSN) %>%
+  distinct() %>% 
+  mutate(step = (row_number() - 1) %/% 100 + 1) # this assigns a value for every 100
+
+log_info(paste0("There are ", count(site_igsns), " sites."))
+
+# create temp site dir
+dir.create(paste0(temp_dir, "/sites"))
+
+# if there are more than 100 sites, then need to break it up for downloading from SESAR
+# this loop grabs 100 sites at a time and walks you through how to download them from SESAR
+for (i in 1:max(site_igsns$step)){
```

Comment:
This goes straight to my downloads

Reply:
Good catch. I added code to give time for the user to manually move the files from the downloads to the temp dir